### ISSUE TITLE
Optimisations fail to build on powerpc64le

### ISSUE THREAD

**Issue by daxtens** [P-H-C/phc-winner-argon2#112]
Hi,

I've tried to build argon2 with optimisations on powerpc64le, and I am hitting a compile failure:
```
dja@dja-builder ~/f/phc-winner-argon2> uname -a
Linux dja-builder 4.2.0-30-generic #35-Ubuntu SMP Fri Feb 19 13:50:54 UTC 2016 ppc64le ppc64le ppc64le GNU/Linux
dja@dja-builder ~/f/phc-winner-argon2> env OPT=TRUE make
rm -f argon2 bench genkat
rm -f libargon2.so libargon2.a kat-argon2* 
rm -f testcase
rm -rf *.dSYM
cd src/ && rm -f *.o
cd src/blake2/ && rm -f *.o
cd kats/ &&  rm -f kat-* diff* run_* make_*
cc -std=c89 -pthread -O3 -Wall -g -Iinclude -Isrc -march=native  src/argon2.c src/core.c src/blake2/blake2b.c src/thread.c src/encoding.c src/opt.c src/run.c -o argon2
cc: error: unrecognized command line option ‘-march=native’
Makefile:75: recipe for target 'argon2' failed
make: *** [argon2] Error 1
```

This is due to powerpc gcc not supporting `-march`. We do have `-mcpu` and `-mtune`, which could be used instead. However, if I change `-march` to `-mcpu`, I get other problems:

```
In file included from src/opt.c:19:0:
src/opt.h:18:23: fatal error: emmintrin.h: No such file or directory
compilation terminated.
```

It looks like you're using Intel intrinsics, meaning that your optimised code would only ever compile on Intel. Is that right?

If so, would it be possible to make `OPT=TRUE` behave differently on platforms other than x86? For example:
  * x86: `-march` and intrinsics
  * other platforms supporting `-march`: `-march` and reference code
  * other platforms supporting `-mcpu`: `-mcpu` and reference code
  * other platforms supporting neither: reference code only

I'm happy to help with compiler flag detection macros or anything else if that would help.

**Comment by sneves** [comment]
You are correct, the optimized version is x86-specific.

**Comment by ranisalt** [comment]
It would be a piece of cake using a different build system as they often have tools to check for compiler options (e.g. CMake's [check_c_compiler_flag](https://cmake.org/cmake/help/v3.0/module/CheckCCompilerFlag.html)) but not so using plain Makefiles.

**Comment by technion** [comment]
Would you mind closing @daxtens  - should be resolved by #113.
